### PR TITLE
DDCE-1064: Authorised official Previous Address Lookup

### PR DESCRIPTION
**Pull request by abhijeetardale**
# DDCE-1064

##Bug fix / New feature (delete as appropriate)

Please include a summary / description of the change and which issue it fixes.  Include any relevant user needs, context or links to other PRs related to this PR (eg. acceptance tests, environment config).

## PR Suggestions
- Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?
- Have you done a visual check of the changes?
- Is there any still commented or unused code? Lingering printlns?
- Have things been implemented in a complicated way?
- Are the test still over the coverage threshold?
- Does the compiler throw a lot of warnings? 
- Have methods and tests been named clearly?
- Were there any changes in the config? Do changes need to be made in app-config-???

## Checklist PR Raiser
 - [x]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [x]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [x]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [x]  I've squashed my commits - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date

## Checklist PR Reviewer
 - [ ]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [ ]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [ ]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [ ]  I've squashed my commits - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date

### CODE REVIEW

**Comment by abhijeetardale** [comment]
Liked PR please
https://github.com/hmrc/charities-registration-journey-tests/pull/150

File: app/viewmodels/authorisedOfficials/AddedOneAuthorisedOfficialHelper.scala [hunk 1]
```diff
@@ -89,9 +89,14 @@ class AddedOneAuthorisedOfficialHelper(index: Index, mode: Mode, countryService:
                   messagePrefix = "authorisedOfficialAddress")
 
   def authOfficialHadPreviousAddressRow: Option[SummaryListRow] =
-    answerPrefix(AuthorisedOfficialPreviousAddressPage(index),
+    answerPrefix(IsAuthorisedOfficialPreviousAddressPage(index),
                  authOfficialRoutes.AuthorisedOfficialPreviousAddressController.onPageLoad(mode, index),
-                 messagePrefix = "authorisedOfficialPreviousAddress")
+                 messagePrefix = "isAddAnotherAuthorisedOfficial")
```
Comment (RafSobol) [inline app/viewmodels/authorisedOfficials/AddedOneAuthorisedOfficialHelper.scala:94]: need to change messages - that labels the summary as 'Add another authorised official'

File: conf/messages.en [hunk 2]
```diff
@@ -982,6 +982,25 @@ authorisedOfficialPreviousAddress.error.required = Select yes if their address h
 
 authorisedOfficialPreviousAddress.checkYourAnswersLabel = Address changed in the last 12 months
 
+# Authorised Official Previous Address Messages
```
Comment (RafSobol) [inline conf/messages.en:985]: both this page and previous ('address changed in last 12 months?') use prefix `authorisedOfficialPreviousAddress`

File: app/viewmodels/authorisedOfficials/AddedOneAuthorisedOfficialHelper.scala (hunk 1 above)
Comment (abhijeetardale) [reply app/viewmodels/authorisedOfficials/AddedOneAuthorisedOfficialHelper.scala:94]: it is naming convetion conflicts between authorisedOfficialPreviousAddress and isAuthorisedOfficialPreviousAddress, doen the rename for cotrollers as well

File: conf/messages.en (hunk 2 above)
Comment (abhijeetardale) [reply conf/messages.en:985]: it is naming convetion conflicts between authorisedOfficialPreviousAddress and isAuthorisedOfficialPreviousAddress, doen the rename for cotrollers as well

File: conf/messages.en [hunk 3]
```diff
@@ -976,11 +976,30 @@ authorisedOfficialAddress.checkYourAnswersLabel = Home address
...
+authorisedOfficialPreviousAddress.selectPage.editAddressLinkText = Enter authorised official’s full previous home address manually
+
+authorisedOfficialPreviousAddress.editPage.title = What is the authorised official’s full previous home address? - Officials and nominees
+authorisedOfficialPreviousAddress.editPage.heading = What is {0}’s full previous home address?
+
+authorisedOfficialPreviousAddress.confirmPage.title = Confirm the authorised official’s previous home address - Officials and nominees
+authorisedOfficialPreviousAddress.confirmPage.heading = Confirm {0}’s previous home address
+authorisedOfficialPreviousAddress.confirmPage.changeLinkText = Change authorised official’s previous home address
+
+authorisedOfficialPreviousAddress.checkYourAnswersLabel = Previous Home address
```
Comment (RafSobol) [inline conf/messages.en:1002]: change to `Previous home address` please (lower-case 'h', both in messages.en and messages.cy)

**Review by RafSobol — APPROVED** [review]

**Pull request merged**